blog/2023-08-15-new-method-to-existing-resource.md

```diff
@@ -0,0 +1,87 @@
+---
+title: Adding purge & aproxSize to the queue resource
+description: 
+authors: 
+  - revitalb
+tags: [cloud-oriented programming, winglang, community]
+hide_table_of_contents: true
+---
+ 
+I love working with the Wing community, and one of my favorite ways to engage is by publishing good first issues that serve as a simple entry point to Wing! I was looking for an issue that could serve as a friendly introduction to Wing's various layers, without overwhelming complexity! Adding a new method to a resource seemed to be a perfect match for these criteria. I posted, as a good first issue, the implementation of the `Queue.Purge` method, and within just 5 minutes I decided to take on the task myself. This felt like the perfect opportunity to dive into the heart of Wing's mechanics and explore how things work behind the scenes.
+
+## Getting Started
+
+Before diving into implementation, it was crucial for me to go through the specifications of the resource I was about to work on. I found all the necessary details in the [Wing spec](https://www.winglang.io/docs/standard-library/cloud/bucket), outlining the method's signature, return type, and arguments. 
+
+## Adding Cloud-Agnostic Interfaces
+
+To add a method to an existing resource, I needed to modify the implementation files of the resource that are located in `wingsdk/src/cloud` folder. The `cloud` folder contains cloud-agnostic code, ensuring that implementations are not tied to any specific cloud provider.
+Under the `cloud` folder, I navigated to `queue.ts` file that contains the cloud-agnostic implementation, added the relevant methods as well as comments and documentations using code suggestions from **co-pilot**!
+
+## Cloud-Specific Implementations
+
+Wing's support for multiple cloud providers means that each new method must be implemented for every supported provider. Now it's time to change folders like `target-awscdk`, `target-sim`, and others, each contain cloud-specific implementation files. 
+Let me give you an example, for cloud provider `target-sim`, I set the length of the `mesages` array to zero:
```

Comment:
Maybe we need some explanation about the sim target since it is not a real cloud